sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala

```diff
@@ -98,10 +98,10 @@ class DateExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
 
   test("datetime function localtimestamp") {
     outstandingTimezonesIds.foreach { zid =>
+      val zoneId = DateTimeUtils.getZoneId(zid)
       val ct = LocalTimestamp(Some(zid)).eval(EmptyRow).asInstanceOf[Long]
-      val t1 = DateTimeUtils.localDateTimeToMicros(
-        LocalDateTime.now(DateTimeUtils.getZoneId(zid)))
-      assert(math.abs(t1 - ct) < 5000)
+      val t1 = DateTimeUtils.localDateTimeToMicros(LocalDateTime.now(zoneId))
+      assert(math.abs(t1 - ct) < 1000000)
```

Comment:
What if the test runs during daylight saving time? So, the diff can be 1 hour.

Reply:
only test it on UTC then?

Reply:
The chance to hit daylight saving time is less than 1/(365*24*3600)..The operation should be finished in 1 second. I can filter "PST" and "LA" if you really want to avoid it. WDYT?

Reply:
There is only a momentary chance of reaching daylight saving time every year.

Reply:
Actually the current_timestamp test has the same DST issue and the test is working pretty well. It should be fine.

Reply:
> If the execution time is 8ms, the chance to hit daylight saving time is less than 8/(365 * 24 * 3600 * 1000)

@gengliangwang This is wrong. You should take into account all runs in forks (not only OSS).

> Actually the current_timestamp test has the same DST issue ...

@cloud-fan Don't think so. Could you explain me, please, how the result of `current_timestamp` is shifted during DST.

FYI, I have already fixed a few tests during DST. You can look for fixes to OSS during LA DST, like https://github.com/apache/spark/pull/27474

Reply:
@MaxGekk since you insist, I created https://github.com/apache/spark/pull/33354 to address your comment.